File: JavaScript solutions/1492. The kth Factor of n.js
/**
 * @param {number} n
 * @param {number} k
 * @return {number}
 */
var kthFactor = function (n, k) {
  const v = [];
  for (let i = 1; i * i <= n; i++) {
    if (i * i == n) v.push(i);
    else if (n % i == 0) {
      v.push(i);
      v.push(n / i);
    }
  }
  v.sort((a, b) => a - b);
  if (k > v.length) return -1;
  return v[k - 1];
};

File: C++ solutions/1492. The kth Factor of n.cpp
class Solution {
public:
    int kthFactor(int n, int k) {
        vector<int>v;
        for(int i=1;i*i<=n;i++){
            if(i*i==n) v.push_back(i);
            else if(n%i==0){
                v.push_back(i);
                v.push_back(n/i);
            }
        }
        sort(v.begin(),v.end());
        if(k>v.size()) return -1;
        return v[k-1];
    }
};